common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -335,6 +327,14 @@ public AsciiString(CharSequence value, int start, int length) {
         }
     }
 
+    public int hashCodeCaseInsensitive() {
+        if (caseInsensitiveHashCode == CASE_INSENSITIVE_HASHER.emptyHashValue() && length() > 0) {
```

Comment:
Yip :) I just don't know if the JIT will substitute the [instanceof](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-6.html#jvms-6.5.instanceof) byte code instruction for an equality check.  However inlining methods (I think it is something like 35 bytes or less) I know it does.